src/python/zquantum/qaoa/estimators.py

```diff
@@ -45,35 +44,49 @@ def get_estimated_expectation_values(
         Returns:
             ExpectationValues: expectation values for each term in the target operator.
         """
-        if alpha > 1 or alpha <= 0:
+        if self.alpha > 1 or self.alpha <= 0:
             raise ValueError("alpha needs to be a value between 0 and 1.")
 
-        if not isinstance(target_operator, IsingOperator):
-            raise TypeError("Operator should be of type IsingOperator.")
-
-        distribution = backend.get_bitstring_distribution(circuit, n_samples=n_samples)
-        expected_values_per_bitstring = {}
+        circuits, operators, shots_per_circuit = zip(
+            *[(e.circuit, e.operator, e.number_of_shots) for e in estimation_tasks]
+        )
+        distributions_list = [
+            backend.get_bitstring_distribution(circuit, n_samples=n_shots)
+            for circuit, n_shots in zip(circuits, shots_per_circuit)
+        ]
 
-        for bitstring in distribution.distribution_dict:
-            expected_value = Measurements(bitstring).get_expectation_values(
-                target_operator
+        return [
+            ExpectationValues(
+                np.array(
+                    _calculate_expectation_value_for_distribution(
+                        distribution, operator, self.alpha
+                    )
+                )
             )
-            expected_values_per_bitstring[bitstring] = expected_value.values[0]
+            for distribution, operator in zip(distributions_list, operators)
+        ]
 
-        sorted_expected_values_per_bitstring_list = sorted(
-            expected_values_per_bitstring.items(), key=lambda item: item[1]
-        )
 
-        cumulative_prob = 0.0
-        cumulative_value = 0.0
+def _calculate_expectation_value_for_distribution(distribution, operator, alpha):
```

Comment:
Should we get in the habit of adding typing for private functions as well?

Reply:
We should.

Reply:
Fixeed in: https://github.com/zapatacomputing/z-quantum-qaoa/pull/44/commits/298309672e513eeb139f12582379ed82e700e38e

Reply:
Still needs a return type

Reply:
🤦 
Fixed in https://github.com/zapatacomputing/z-quantum-qaoa/pull/44/commits/43793d77bea62560d6dc86f528741972aaa739a3